apisix/plugins/limit-count/init.lua

```diff
@@ -300,9 +327,28 @@ function _M.rate_limit(conf, ctx)
         return 500, {error_msg = "failed to limit count"}
     end
 
+    local reset
+    if remaining == conf.count - 1 then
+        -- set an end time
+        local end_time = ngx_time() + conf.time_window
+        -- save to lrucache by key
+        reset = conf.time_window
+        local end_time_obj = get_end_time(conf,key)
```

Comment:
By default, the client IP will be used as key, which means there will be a great number of keys per conf. The size of lrucache is limited.

There come two solutions:
1. store the end time info in the actual backend like redis cluster.
2. write a new cache with dynamical expire support to store the end time info, but this way is not memory-effective.

Personally, I prefer solution 1.

Reply:
solution 1 does not work either. Because by default limit count will not use Redis as the backend. So we can't save it to Redis.

Reply:
By default, There use `resty.limit.count` this package for limit traffic. And I read the code of Resty Limit Count package using `ngx.shared.DICT` to save data. Can I use this?

Reply:
I'm using `ngx.shared` with expire time instead.

Reply:
The `ngx.shared` only works with the default backend which also uses `ngx.shared`. If we use other backends, only one instance will execute `remaining == conf.count - 1` and the end time can only be found in that instance.

Reply:
Fixed, you can review it again.